aisdc/postprocessing/generate_report.py

```diff
@@ -0,0 +1,283 @@
+import json
+import os
+import pprint
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+class AnalysisModule:
+    """
+    Wrapper module for metrics analysis modules
+    """
+
+    def process_dict(self):
+        """
+        Function that produces a risk summary output based on analysis in this module
+        """
+        raise NotImplementedError()
+
+    def __str__(self):
+        raise NotImplementedError()
+
+
+class FinalRecommendationModule(AnalysisModule):
+    """
+    Module that generates the first layer of a recommendation report
+    """
+
+    def __init__(self, report: dict):
+        self.P_VAL_THRESH = 0.1
+        self.MEAN_AUC_THRESH = 0.65
+
+        self.SVM_WEIGHTING_SCORE = 5
```

Comment:
need some thought of how to encode these in a general way as they could easily become cumbersome -- no need to change now though.

Reply:
Agreed - some kind of dictionary (e.g. another JSON file) with default parameters was my initial thought, not urgent for now though